### PR TITLE
Update rangeslider to TypeScript

### PR DESCRIPTION
**Pull request by berkbaski**

### CODE REVIEW

File: ui/rangeslider/rangeslider.ios.ts [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+      }
+    };
+    Object.assign(this._ios, ios);
+
+    const valueChangeHandler = () => {
+      self.onValueChange?.(self.value);
+    };
+    self.nativeObject.addJSTarget(valueChangeHandler, UIControlEvents.valueChanged);
+
+    const EventFunctions = {
```
Comment (cenkce) [inline ui/rangeslider/rangeslider.ios.ts:79]: bunu kaldiralim gerek yok

File: ui/rangeslider/rangeslider.ios.ts [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+      [RangeSliderEvents.ValueChange]: (e) => {
+        self.onValueChange(e);
+      }
+    };
+    eventCallbacksAssign(this, EventFunctions);
```
Comment (cenkce) [inline ui/rangeslider/rangeslider.ios.ts:84]: buna da gerek yok

File: ui/rangeslider/rangeslider.ios.ts [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+      set showsThumbImageShadow(value: boolean) {
+        self.nativeObject.showsThumbImageShadow = value;
+      },
+      applyThumbViewChanges: () => {
+        self.nativeObject.applyThumbViewChanges();
+      }
+    };
+    Object.assign(this._ios, ios);
+
+    const valueChangeHandler = () => {
```
Comment (cenkce) [inline ui/rangeslider/rangeslider.ios.ts:74]: `this.emit(RangeSliderEvents.ValueChange)` ekleyelim

File: ui/rangeslider/rangeslider.android.ts [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+      },
+      set outerTrackWeight(value: number) {
+        self._outerTrackWeight = value;
+        self.nativeObject.setBarWeight(AndroidUnitConverter.dpToPixel(self._outerTrackWeight));
+      }
+    };
+    Object.assign(this._android, android);
+
+    const EventFunctions = {
+      [RangeSliderEvents.ValueChange]: (e) => {
```
Comment (cenkce) [inline ui/rangeslider/rangeslider.android.ts:77]: `EventFunctions`  bunu kaldiralim

File: ui/rangeslider/rangeslider.android.ts [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+      get outerTrackWeight(): number {
+        return self._outerTrackWeight;
+      },
+      set outerTrackWeight(value: number) {
+        self._outerTrackWeight = value;
+        self.nativeObject.setBarWeight(AndroidUnitConverter.dpToPixel(self._outerTrackWeight));
+      }
+    };
+    Object.assign(this._android, android);
+
```
Comment (cenkce) [inline ui/rangeslider/rangeslider.android.ts:75]: `this.nativeObject.setOnValueChange(() => {
  this.emit(RangeSliderEvents.ValueChange);
  this.onValueChange?.(e);
})`
ekleyelim

**Review by cenkce — CHANGES_REQUESTED** [review]

**Comment by berkbaski** [comment]
Could you check it again? @cenkce

File: ui/rangeslider/rangeslider.ios.ts [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+    Object.assign(this._ios, _ios);
+    Object.assign(this, restParams);
+
+    const valueChangeHandler = () => {
+      self.onValueChange?.(self.value);
+      self.emit(RangeSliderEvents.ValueChange, self.value);
+    };
+    self.nativeObject.addJSTarget(valueChangeHandler, UIControlEvents.valueChanged);
+
+    this.setParams(params);
```
Comment (cenkce) [inline ui/rangeslider/rangeslider.ios.ts:81]: `    this.setParams(params);
` kaldiralim

File: ui/rangeslider/rangeslider.ios.ts [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+  }
+
+  get onValueChange(): (value: number[]) => void {
+    return this._onValueChange;
+  }
+  set onValueChange(value: (value: number[]) => void) {
+    this._onValueChange = value;
+  }
+
+  setParams(params: Partial<IRangeSlider>) {
```
Comment (cenkce) [inline ui/rangeslider/rangeslider.ios.ts:177]: setParams kaldiralim

File: ui/rangeslider/rangeslider.ios.ts [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+    for (const param in params) {
+      if (param === 'ios' || param === 'android') {
+        // TODO Recheck after build
+        this.setOSSpecificParams(params[param], param);
+      } else {
+        this[param] = params[param];
+      }
+    }
+  }
+  setOSSpecificParams(params: Partial<IRangeSlider>, key: string) {
```
Comment (cenkce) [inline ui/rangeslider/rangeslider.ios.ts:187]: setOSSpecificParams kaldiralim

**Review by cenkce — CHANGES_REQUESTED** [review]

**Comment by berkbaski** [comment]
I added `WithMobileOSProps` Could you recheck again? @cenkce

**Pull request merged**